src/conn/conn_compact.c

```diff
@@ -179,14 +181,14 @@ __compact_server(void *arg)
     WT_STAT_CONN_SET(session, background_compact_running, 0);
 
     WT_ERR(__wt_metadata_cursor_close(session));
```

Comment:
Not the subject of this review, but you don't need (and probably don't want) this call to close the metadata cursor. 

The standard way to use a metadata cursor is to call `__wt_metadata_cursor()` to get a metadata cursor and `__wt_metadata_cursor_release()` when you are done with it, which is what is happening in the loop, above. These functions will open and close a metadata cursor for you. 

See, for example, how these are used in the main loop of `__checkpoint_flush_tier()`

Reply:
Thank you for spotting this, fixed in ab8348968b732d7ceae31d77ced88da0bfc6f971.

Reply:
Sorry. I should have looked at the code more carefully. I think you'll need to put the cursor release after the `err:` label because there are calls to `WT_ERR()` in the body of the loop in places where the cursor is open.

Reply:
Yes, we should move this down, thanks again, fixed in e39eafde491166561c9addbf1ac8a3ebc4cf7200.

Reply:
> I haven't thought it through, but my instinct would be to see if the code for getting the next item from the metadata table can be turned into a separate helper function.

I completely agree! @Sean04 and I discussed this, a function to "get the next uri to compact" can be created easily here. This could be done in WT-11343.